ninja/operation.py

```diff
@@ -207,7 +207,11 @@ def _run_authentication(self, request: HttpRequest) -> Optional[HttpResponse]:
                 if is_async_callable(callback) or getattr(callback, "is_async", False):
                     result = callback(request)
                     if inspect.iscoroutine(result):
-                        result = async_to_sync(callback)(request)
+
+                        async def await_result(cor: Coroutine) -> Any:
+                            return await cor
+
+                        result = async_to_sync(await_result)(result)
                 else:
                     result = callback(request)
```

Comment:
In the `try` block,
```python
                result = callback(request)
                if inspect.iscoroutine(result):
                    result = async_to_sync(callback)(request)

                    async def await_result(cor: Coroutine) -> Any:
                        return await cor

                    result = async_to_sync(await_result)(result)
```
would not be enough since `result` is created the same way in all cases and we may want to await `result` independently of the properties of `callback`?